Mathlib/Computability/Chomsky.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2024 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak
+-/
+import Mathlib.Computability.ContextFreeGrammar
+import Mathlib.Computability.DFA
+import Mathlib.Data.Fintype.Prod
+
+/-!
+# Chomsky hierarchy
+
+This file attempts to capture the Chomsky hierarchy. We want to show inclusions between classes
+of languages: Regular ⊆ ContextFree ⊆ ContextSensitive ⊆ RecursivelyEnumerable
```

Comment:
Looks good so far, I just suggest making TODOs of the remaining items here if possible.